sky/clouds/kubernetes.py

```diff
@@ -109,17 +109,19 @@ def _unsupported_features_for_resources(
     def max_cluster_name_length(cls) -> Optional[int]:
         return cls._MAX_CLUSTER_NAME_LEN_LIMIT
 
-    @classmethod
-    def regions(cls) -> List[clouds.Region]:
-        return cls._regions
-
     @classmethod
     def regions_with_offering(cls, instance_type: Optional[str],
                               accelerators: Optional[Dict[str, int]],
                               use_spot: bool, region: Optional[str],
                               zone: Optional[str]) -> List[clouds.Region]:
-        # No notion of regions in Kubernetes - return a single region.
-        return cls.regions()
+        allowed_contexts = skypilot_config.get_nested(
+            ('kubernetes', 'allowed_contexts'), None)
+        if allowed_contexts is None:
+            return cls._regions
```

Comment:
I am thinking to still using the same semantics as the original when `allowed_contexts` is not specified. In this way, single k8s clusters will not need to understand another concept. Wdyt?

Reply:
I think there's three main benefits of switching to changing region to use context name:

1. Keeps our YAMLs/CLI consistent irrespective of whether `allowed_contexts` is used.  For example, say a user wrote a YAML like this when they were using `allowed_contexts: [dev, staging, prod]`:
    ```bash
     resources:
       cloud: kubernetes
       region: dev
     ```
    Now say they removed allowed_contexts or shared this with a colleague without `allowed_contexts` set.  They would need to update the YAML to `region: kubernetes` or remove the region flag to make this work. Also, they would lose the "run this YAML only on dev cluster" directive from the YAML.
2. Helps https://github.com/skypilot-org/skypilot/issues/3461. This is a common problem if the user switches contexts (even though they may use one at a time).
3. Showing the context name under region during `sky launch` helps reaffirm a) the cluster your job will run on and b) SkyPilot can switch between contexts now.

Even for single k8s users, using a concrete region name instead of `kubernetes` might be easier to understand. wdyt?

Reply:
That is a good point! I have changed the region to always show the context.